ros_ign_gazebo/src/create.cpp

```diff
@@ -98,27 +98,49 @@ int main(int _argc, char ** _argv)
       req.set_sdf(xmlStr);
     } else {
       RCLCPP_ERROR(
-        ros2_node->get_logger(), "Failed to get XML from param [%s].",
-        FLAGS_param.c_str());
+        ros2_node->get_logger(), "Failed to get XML from param [%s].", FLAGS_param.c_str());
       return -1;
     }
   } else if (!FLAGS_string.empty()) {  // string
     req.set_sdf(FLAGS_string);
+  } else if (!FLAGS_topic.empty()) {  // topic
+    const auto timeout = std::chrono::seconds(1);
+    std::promise<std::string> xml_promise;
+    std::shared_future<std::string> xml_future(xml_promise.get_future());
+
+    std::function<void(const std_msgs::msg::String::SharedPtr)> fun =
+      [&xml_promise](const std_msgs::msg::String::SharedPtr msg) {
+        xml_promise.set_value(msg->data);
+      };
+
+    rclcpp::executors::SingleThreadedExecutor executor;
+    executor.add_node(ros2_node);
+    rclcpp::Subscription<std_msgs::msg::String>::SharedPtr description_subs;
+    // Transient local is similar to latching in ROS 1.
+    description_subs = ros2_node->create_subscription<std_msgs::msg::String>(
+      FLAGS_topic, rclcpp::QoS(1).transient_local(), fun);
+
+    rclcpp::FutureReturnCode future_ret;
+    do {
+      RCLCPP_INFO(ros2_node->get_logger(), "Waiting messages on topic [%s].", FLAGS_topic.c_str());
+      future_ret = executor.spin_until_future_complete(xml_future, timeout);
+    } while (rclcpp::ok() && future_ret != rclcpp::FutureReturnCode::SUCCESS);
+
+    if (future_ret == rclcpp::FutureReturnCode::SUCCESS) {
+      RCLCPP_INFO(ros2_node->get_logger(), xml_future.get());
```

Comment:
This could be huge depending on the model being spawned. Consider omitting this info here.

Reply:
Yes you are right!